src/odemis/acq/test/fastem_test.py

```diff
@@ -109,6 +115,362 @@ def test_overview_acquisition(self):
+        cls.mppc = model.getComponent(role="mppc")
+        cls.multibeam = model.getComponent(role="multibeam")
+        cls.descanner = model.getComponent(role="descanner")
+        cls.stage = model.getComponent(role="stage")  # TODO replace with stage-scan when ROA conversion method available
+        cls.stage.reference({"x", "y"}).result()
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.backend_was_running:
+            return
+        test.stop_backend()
+
+    def setUp(self):
+        if self.backend_was_running:
+            self.skipTest("Running backend found")
+
+    def test_acquire_ROA(self):
+        """Acquire a small mega field image."""
+        x_fields = 3
+        y_fields = 4
+        res_x, res_y = self.multibeam.resolution.value  # single field size
+        px_size_x, px_size_y = self.multibeam.pixelSize.value
+        coordinates = (0, 0, res_x * px_size_x * x_fields, res_y * px_size_y * y_fields)  # in m
+        roc = fastem.FastEMROC("roc_name", coordinates)
+        roa_name = time.strftime("test_megafield_id-%Y-%m-%d-%H-%M-%S")
+        roa = fastem.FastEMROA(roa_name,
+                               coordinates,
+                               roc,
+                               self.asm,
+                               self.multibeam,
+                               self.descanner,
+                               self.mppc
+                               )
+        path_storage = os.path.join(datetime.today().strftime('%Y-%m-%d'), "test_project_megafield")
+        f = fastem.acquire(roa, path_storage, self.multibeam, self.descanner, self.mppc, self.stage)
+        data, e = f.result()
+
+        self.assertIsNone(e)  # check no exceptions were returned
+        # check data returned contains the correct number of field images
+        self.assertEqual(len(data), x_fields * y_fields)
+        self.assertIsInstance(data[(0, 0)], model.DataArray)
+
+    def test_progress_ROA(self):
+        """Check if some progress is reported between the field images acquired for the ROA (megafield)."""
+        x_fields = 2
+        y_fields = 3
+        res_x, res_y = self.multibeam.resolution.value  # single field size
+        px_size_x, px_size_y = self.multibeam.pixelSize.value
+        coordinates = (0, 0, res_x * px_size_x * x_fields, res_y * px_size_y * y_fields)  # in m
+        roc = fastem.FastEMROC("roc_name", coordinates)
+        roa_name = time.strftime("test_megafield_id-%Y-%m-%d-%H-%M-%S")
+        roa = fastem.FastEMROA(roa_name,
+                               coordinates,
+                               roc,
+                               self.asm,
+                               self.multibeam,
+                               self.descanner,
+                               self.mppc
+                               )
+        path_storage = os.path.join(datetime.today().strftime('%Y-%m-%d'), "test_project_progress")
+
+        self.start = None
+        self.end = None
```

Comment:
Here it looks like these are unused, can you comment on why they are necessary and that they are updated in on_progress_update